nipype/interfaces/niftyreg/reg.py

```diff
@@ -0,0 +1,439 @@
+    flo_low_val = traits.Float(desc='Lower threshold value on floating image',
+                               argstr='-floLowThr %f')
+    # Upper threshold on floating image
+    flo_up_val = traits.Float(desc='Upper threshold value on floating image',
+                              argstr='-floUpThr %f')
+    # Platform to use
+    platform_val = traits.Int(desc='Platform index',
+                              argstr='-platf %i')
+    # Platform to use
+    gpuid_val = traits.Int(desc='Device to use id',
+                           argstr='-gpuid %i')
+    # Verbosity off
+    verbosity_off_flag = traits.Bool(argstr='-voff',
+                                     desc='Turn off verbose output')
+    # Set the number of omp thread to use
+    omp_core_val = traits.Int(desc='Number of openmp thread to use',
+                              argstr='-omp %i')
+
+    # Affine output transformation matrix file
+    aff_file = File(genfile=True,
+                    desc='The output affine matrix file',
+                    argstr='-aff %s')
+    # Result warped image file
+    res_file = File(genfile=True,
+                    desc='The affine transformed floating image',
+                    argstr='-res %s')
+
+
+class RegAladinOutputSpec(TraitedSpec):
+    """ Output Spec for RegAladin. """
+    aff_file = File(desc='The output affine file')
+    res_file = File(desc='The output transformed image')
+    desc = 'Output string in the format for reg_average'
+    avg_output = traits.String(desc=desc)
+
+
+class RegAladin(NiftyRegCommand):
+    """Interface for executable reg_aladin from NiftyReg platform.
+
+    Block Matching algorithm for symmetric global registration.
+    Based on Modat et al., "Global image registration using
+    asymmetric block-matching approach"
+    J. Med. Img. 1(2) 024003, 2014, doi: 10.1117/1.JMI.1.2.024003
+
+    For source code, see https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg
+
+    Examples
+    --------
+    >>> from nipype.interfaces import niftyreg
+    >>> node = niftyreg.RegAladin()
+    >>> node.inputs.ref_file = 'im1.nii'
+    >>> node.inputs.flo_file = 'im2.nii'
+    >>> node.inputs.rmask_file = 'mask.nii'
+    >>> node.inputs.omp_core_val = 4
+    >>> node.cmdline  # doctest: +ELLIPSIS +ALLOW_UNICODE
+    'reg_aladin -aff .../im2_aff.txt -flo im2.nii -omp 4 -ref im1.nii \
+-res .../im2_res.nii.gz -rmask mask.nii'
+
+    """
+    _cmd = get_custom_path('reg_aladin')
+    input_spec = RegAladinInputSpec
+    output_spec = RegAladinOutputSpec
+
+    def _gen_filename(self, name):
+        if name == 'aff_file':
+            return self._gen_fname(self.inputs.flo_file,
+                                   suffix='_aff', ext='.txt')
+        if name == 'res_file':
+            return self._gen_fname(self.inputs.flo_file,
+                                   suffix='_res', ext='.nii.gz')
+        return None
+
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
```

Comment:
this function can be reduced to:

```
 def _list_outputs(self):
     outputs = super(RegAladin, self)._list_outputs()
     # Make a list of the linear transformation file and the input image
     aff = os.path.abspath(outputs['aff_file'])
     flo = os.path.abspath(self.inputs.flo_file)
     outputs['avg_output'] = '%s %s' % (aff, flo)
     return outputs
```

it seems that the `avg_output` here is combining things to make it easy for some other interface. this breaks the dataflow paradigm as the next interface will not be able to use such a string to determine the hash of the input properly. it would be better to do such a merge separately in a workflow rather than as an output.